src/Uno.UI/UI/Xaml/Controls/TreeView/TreeViewViewModel.Nested.cs

```diff
@@ -0,0 +1,280 @@
+						{
+							var item = listControl.ItemFromNode(node);
+							if (item != null)
+							{
+								selectedItems.Insert(index, item);
+							}
+						}
+					}
+				}
+			}
+
+			internal void RemoveAtCore(int index)
+			{
+				GetVectorInnerImpl().RemoveAt(index);
+
+				// Keep SelectedItems and SelectedNodes in sync
+				var viewModel = m_viewModel;
+				if (viewModel != null)
+				{
+					var selectedItems = viewModel.SelectedItems;
+					if (Count != selectedItems.Count)
+					{
+						selectedItems.RemoveAt(index);
+					}
+				}
+			}
+
+			//void InsertAt(int index, TreeViewNode node)
+			//{
+			//	if (!Contains(node))
+			//	{
+			//		// UpdateSelection will call InsertAtCore
+			//		UpdateSelection(node, TreeNodeSelectionState.Selected);
+			//	}
+			//}
+
+
+			//#pragma region SelectedTreeNodeVector
+
+			//		typedef typename VectorOptionsFromFlag<winrt::TreeViewNode, MakeVectorParam<VectorFlag::Observable, VectorFlag::DependencyObjectBase>()> SelectedTreeNodeVectorOptions;
+
+			//class SelectedTreeNodeVector :
+
+			//	public ReferenceTracker<
+			//	SelectedTreeNodeVector,
+			//	reference_tracker_implements_t<typename SelectedTreeNodeVectorOptions::VectorType>::type,
+			//	typename TreeViewNodeVectorOptions::IterableType,
+			//	typename TreeViewNodeVectorOptions::ObservableVectorType>,
+			//    public TreeViewNodeVectorOptions::IVectorOwner
+			//{
+			//    Implement_Vector_Read(SelectedTreeNodeVectorOptions)
+
+			//private:
+			//    
+			//}
+			//    }
+
+			//public:
+			//   
+
+
+
+
+
+			//void SetAt(unsigned int index, winrt::TreeViewNode const& node)
+			//{
+			//	RemoveAt(index);
+			//	InsertAt(index, node);
+			//}
+
+			//void RemoveAt(unsigned int index)
+			//{
+			//	auto inner = GetVectorInnerImpl();
+			//	auto oldNode = winrt::get_self<TreeViewNode>(inner->GetAt(index));
+			//	// UpdateNodeSelection will call RemoveAtCore
+			//	UpdateSelection(*oldNode, TreeNodeSelectionState::UnSelected);
+			//}
+
+			//void RemoveAtEnd()
+			//{
+			//	RemoveAt(Size() - 1);
+			//}
+
+			//void ReplaceAll(winrt::array_view<winrt::TreeViewNode const> nodes)
+			//{
+			//	Clear();
+
+			//	for (auto const&node : nodes)
+			//        {
+			//		Append(node);
+			//	}
+			//}
+
+			//void Clear()
+			//{
+			//	while (Size() > 0)
+			//	{
+			//		RemoveAtEnd();
+			//	}
+			//}
+
+			//bool Contains(winrt::TreeViewNode const& node)
+			//{
+			//	uint32_t index;
+			//	return GetVectorInnerImpl()->IndexOf(node, index);
+			//}
+
+		}
+
+
+
+		//#pragma endregion
+
+		//// Similiar to SelectedNodesVector above, we need to make decisions before the item is inserted or removed.
+		//// we can't use vector change events because the event already happened when event hander gets called.
+		//#pragma region SelectedItemsVector
+
+		//typedef typename VectorOptionsFromFlag<winrt::IInspectable, MakeVectorParam<VectorFlag::Observable, VectorFlag::DependencyObjectBase>()> SelectedItemsVectorOptions;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)